### PR TITLE
Changes in EF Geth Release 1.4.14

### PR DESCRIPTION
**Pull request by splix**

### CODE REVIEW

**Comment by splix** [comment]
@elaineo @ericsomdahl @whatisgravity please review

**Comment by splix** [comment]
Absolutely agree about careful checking all this changes

File: build/ci.go [hunk 1]
```diff
@@ -227,6 +227,9 @@ func doTest(cmdline []string) {
 
 	// Run the actual tests.
 	gotest := goTool("test")
+	// Test a single package at a time. CI builders are slow
+	// and some tests run into timeouts under load.
+	gotest.Args = append(gotest.Args, "-p", "1")
```
Comment (splix) [inline build/ci.go]: I guess. Not sure we actually use it for our CI tests, but could be useful for manual testing too

**Comment by elaineo** [comment]
Okay, it looks like the 1.4.14 changes are intended to be used with 1.4.16. I think 1.4.13 is safe to merge. It looks like geth nodes are under attack on eth again.

File: build/ci.go (hunk 1 above)
Comment (ericsomdahl) [inline build/ci.go]: I have started modifying this file to support our build process.  This particular change is of no consequence for us

File: cmd/utils/cmd.go [hunk 2]
```diff
@@ -52,10 +53,16 @@ func openLogFile(Datadir string, filename string) *os.File {
 // is redirected to a different file.
 func Fatalf(format string, args ...interface{}) {
 	w := io.MultiWriter(os.Stdout, os.Stderr)
-	outf, _ := os.Stdout.Stat()
-	errf, _ := os.Stderr.Stat()
-	if outf != nil && errf != nil && os.SameFile(outf, errf) {
-		w = os.Stderr
+	if runtime.GOOS == "windows" {
+		// The SameFile check below doesn't work on Windows.
```
Comment (ericsomdahl) [inline cmd/utils/cmd.go]: Hopefully this does not impact us given our checked history with Windows stream piping :)

File: core/blockchain.go [hunk 3]
```diff
@@ -95,10 +95,11 @@ type BlockChain struct {
 	currentBlock     *types.Block // Current head of the block chain
 	currentFastBlock *types.Block // Current head of the fast-sync chain (may be above the block chain!)
 
-	bodyCache    *lru.Cache // Cache for the most recent block bodies
-	bodyRLPCache *lru.Cache // Cache for the most recent block bodies in RLP encoded format
-	blockCache   *lru.Cache // Cache for the most recent entire blocks
-	futureBlocks *lru.Cache // future blocks are blocks added for later processing
+	stateCache   *state.StateDB // State database to reuse between imports (contains state cache)
```
Comment (ericsomdahl) [inline core/blockchain.go]: the content of this file is a repetition of the 1.4.13 PR

**Comment by ericsomdahl** [comment]
The changes from 1.4.13 PR are also showing in this PR as changes.  Perhaps this will be a test of the smarts of git merging... we may need to rebase this branch against develop before merge

**Pull request merged**